c10/util/Optional.h

```diff
@@ -9,6 +9,8 @@
 // up builds.
 
 namespace c10 {
+
+#if !defined(FBCODE_CAFFE2) && !defined(C10_NODEPRECATED)
```

Comment:
is this bc-breaking? if so, needs to be labeled as such. (it doesn't look like C10_NODEPRECATED is defined anywhere, but perhaps it's defined non-greppably)

Reply:
`C10_NODEPRECATED` is not defined and this is in OSS, so `FBCODE_CAFFE2` is not defined, so the API is still available; thus, it's not bc-breaking.